requirements/extra-ml-requirements.txt

```diff
@@ -36,8 +36,6 @@ onnxruntime
 tf2onnx
 # Required by mlflow.spark
 pyspark
-# Required by mlflow.shap
-shap
```

Comment:
We should reintroduce `shap` as a requirement here; I had previously removed it because the legacy PyPI resolver complained, but this isn't an issue now that we're using the current PyPI resolver.

Reply:
Did the tests complain before or after we removed the legacy resolver?

Reply:
Before. Let me try it now :)